salt/config/__init__.py

```diff
@@ -944,8 +944,8 @@ def _gather_buffer_space():
     # Always generate minion id in lowercase.
     'minion_id_lowercase': bool,
 
-    # By default '', suppress this single domain (example 'foo.org') in a generated minion id.
-    'remove_domain': six.string_types,
+    # Remove this single domain (e.g. 'foo.org') in a generated minion id.
+    'minion_id_remove_domain': six.string_types,
```

Comment:
Can be also `False`. In case of `False` it equals to an empty string, and in case of `True` it can go and remove any domain. Just suggestion.